rotkehlchen/globaldb/handler.py

```diff
@@ -208,6 +209,140 @@ def add_asset(
                 f'Failed to add asset {asset_id} into the assets table due to {str(e)}',
             ) from e
 
+    @staticmethod
+    def retrieve_assets(filter_query: 'AssetsFilterQuery') -> Tuple[List[Dict[str, Any]], int]:
+        """Returns a tuple that contains a list of assets and a count of those
+        assets that match the filter query.
+        May raise:
+        - DeserializationError
+        """
+        assets_list = []
+        query, bindings = filter_query.prepare()
+        parent_query = """
+        SELECT A.identifier AS identifier, A.type, B.address, B.decimals, A.name, C.symbol, C.started, C.forked, C.swapped_for, C.coingecko, C.cryptocompare, B.protocol, B.chain, B.token_kind
+        FROM assets as A
+        JOIN common_asset_details AS C ON C.identifier = A.identifier
+        LEFT JOIN evm_tokens as B ON B.identifier = A.identifier
+        """  # noqa: E501
+        query = f'SELECT * FROM ({parent_query}) ' + query
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            for entry in cursor:
+                asset_type = AssetType.deserialize_from_db(entry[1])
+                data = {
+                    'identifier': entry[0],
+                    'type': str(asset_type),
+                    'name': entry[4],
+                }
+                # for evm tokens and crypto assets
+                common_data = {
+                    'symbol': entry[5],
+                    'started': entry[6],
+                    'swapped_for': entry[8],
+                    'forked': entry[7],
+                    'cryptocompare': entry[10],
+                    'coingecko': entry[9],
+                }
+                if asset_type == AssetType.FIAT:
+                    data.update({
+                        'symbol': entry[5],
+                        'started': entry[6],
+                    })
+                elif asset_type == AssetType.EVM_TOKEN:
+                    data.update({
+                        'address': string_to_evm_address(entry[2]),
```

Comment:
Do you need the `string_to_evm_address`?

Reply:
nah. removing